src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500NameEncoder.cs

```diff
@@ -567,5 +582,70 @@ private static int ExtractValue(ReadOnlySpan<char> chars, Span<char> destination
 
             return written;
         }
+
+        private static Dictionary<string, EncodingRules> CreateEncodingRulesLookup()
+        {
+            const int LookupDictionarySize = 27;
+            Dictionary<string, EncodingRules> lookup = new(LookupDictionarySize, StringComparer.Ordinal)
+            {
+                { Oids.EmailAddress, EncodingRules.IA5String },
+                { Oids.CommonName, EncodingRules.DirectoryString },
+                { Oids.Surname, EncodingRules.DirectoryString },
+                { Oids.GivenName, EncodingRules.DirectoryString },
+                { Oids.Initials, EncodingRules.DirectoryString },
+                { Oids.GenerationQualifier, EncodingRules.DirectoryString },
+                { Oids.DnQualifier, EncodingRules.PrintableString },
```

Comment:
One could argue (but one isn't, really) that we should include id-at-uniqueIdentifier, mapped to EncodingRules.Fail, since we don't have a way to write BIT STRING values with this API.  Instead, we'll just gibberish it up.

Reply:
If we want to be closer to what Windows does, "permit it" is the "right thing". I think I am okay with that. This is a little tricky to balance as in some places we are saying "Windows is wrong here" and in other places we are trying to be more like Windows.